### ISSUE TITLE
v0.23.0 compatibility issue with IE7 and IE8

### ISSUE THREAD

**Issue by nskate** [gregjacobs/Autolinker.js#144]
The new v0.23.0+ utilizes the filter() function, which is not compatible with IE7 and IE8 since those browsers don't have that function.  Would be great for IE7 and IE8 compatibility version.

**Comment by simison** [comment]
Meanwhile you could add this polyfill to your app, [via Mozilla](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter#Polyfill):

```javascript
if (!Array.prototype.filter) {
  Array.prototype.filter = function(fun/*, thisArg*/) {
    'use strict';

    if (this === void 0 || this === null) {
      throw new TypeError();
    }

    var t = Object(this);
    var len = t.length >>> 0;
    if (typeof fun !== 'function') {
      throw new TypeError();
    }

    var res = [];
    var thisArg = arguments.length >= 2 ? arguments[1] : void 0;
    for (var i = 0; i < len; i++) {
      if (i in t) {
        var val = t[i];

        // NOTE: Technically this should Object.defineProperty at
        //       the next index, as push can be affected by
        //       properties on Object.prototype and Array.prototype.
        //       But that method's new, and collisions should be
        //       rare, so use the more-compatible alternative.
        if (fun.call(thisArg, val, i, t)) {
          res.push(val);
        }
      }
    }

    return res;
  };
}
```

**Comment by gregjacobs** [comment]
Ack, good call. Sorry about that. Will replace with a standard loop.

**Comment by gregjacobs** [comment]
Fixed in 0.25.0. Thanks for reporting!
